packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -30,6 +30,8 @@ import { AddItemButton as DefaultAddItemButton } from './AddItemButton';
 import { RemoveItemButton as DefaultRemoveItemButton } from './RemoveItemButton';
 import { ReOrderButtons as DefaultReOrderButtons } from './ReOrderButtons';
 
+let initialDefaultValue = {};
```

Comment:
I'm very scared of that type of global values - it will persist across the entire app. We'll see side effects in tests. 

Can't you find an alternative fix that doesn't rely on a global? Why not a `ref`?